root/static/scripts/release-editor/fields.js

```diff
@@ -366,9 +367,9 @@ class Medium {
             }
         });
 
-        this.needsRecordings = this.tracks.any("needsRecording");
-        this.hasTrackInfo = this.tracks.all("hasNameAndArtist");
-        this.hasVariousArtistTracks = this.tracks.any("hasVariousArtists");
+        this.needsRecordings = !self.tracksUnknownToUser() && this.tracks.any("needsRecording");
```

Comment:
The problem here and below is that these were observables, but are being changed to either `false` (a literal value that will never change), or the `this.tracks.any` observable as it was (which will never be affected by `tracksUnknownToUser` if it changes), since `x && y` just returns either the first falsey value or the last truthy value.

To clarify, this constructor only runs once when the medium is created, not whenever the observables change. If you want a "live" observable that updates when other observables change, you must use a computed observable:

```JS
this.needsRecordings = ko.computed(function () {
  return !self.tracksUnknownToUser() && self.tracks().any(t => t.needsRecording());
});
```

Note that changing `this.tracks.any("needsRecording")` (which actually returns a computed observable) to calling the builtin `any` directly on the unwrapped array is also important.